zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -605,7 +321,7 @@ public DataWatchRegistration(Watcher watcher, String clientPath) {
 
         @Override
         protected Map<String, Set<Watcher>> getWatches(int rc) {
-            return watchManager.dataWatches;
+            return getWatchManager().dataWatches;
```

Comment:
dataWatches -> getDataWatches() 
otherwise we are breaking the incapsulation you are introducing

Reply:
good point. Addressing...